tests/platform_tests/api/test_sfp.py

```diff
@@ -322,16 +414,28 @@ def test_get_transceiver_threshold_info(self, duthosts, enum_rand_one_per_hwsku_
         skip_release_for_platform(duthost, ["202012"], ["arista", "mlnx"])
 
         for i in self.sfp_setup["sfp_test_port_indices"]:
+            info_dict = sfp.get_transceiver_info(platform_api_conn, i)
+
+            if not self.is_xcvr_optical(info_dict):
+                logger.info("test_get_transceiver_threshold_info: Skipping transceiver {} (not applicable for this transceiver type)".format(i))
+                continue
+
             thold_info_dict = sfp.get_transceiver_threshold_info(platform_api_conn, i)
             if self.expect(thold_info_dict is not None, "Unable to retrieve transceiver {} threshold info".format(i)):
                 if self.expect(isinstance(thold_info_dict, dict), "Transceiver {} threshold info appears incorrect".format(i)):
                     actual_keys = thold_info_dict.keys()
 
-                    missing_keys = set(self.EXPECTED_XCVR_THRESHOLD_INFO_KEYS) - set(actual_keys)
+                    expected_keys = self.EXPECTED_XCVR_THRESHOLD_INFO_KEYS
```

Comment:
Also, please rename EXPECTED_XCVR_THRESHOLD_INFO_KEYS to something like EXPECTED_XCVR_COMMON_THRESHOLD_INFO_KEYS or so.

Reply:
Done